src/databases/Silo/avtSiloFileFormat.C

```diff
@@ -10326,7 +10328,7 @@ avtSiloFileFormat::ReadInConnectivity(vtkUnstructuredGrid *ugrid,
         int effective_vtk_zonetype = vtk_zonetype;
         int effective_shapesize = shapesize;
 
-        if (vtk_zonetype < 0 && vtk_zonetype != -2)
+        if (vtk_zonetype < 0 && vtk_zonetype != VTK_POLYHEDRON)
```

Comment:
```suggestion
        if (vtk_zonetype < 0)
```

With your change, I don't think the `&&` part is needed anymore. That was to handle cases where the magic value collides with `<0` test.